examples/media/src/widgets/Articles.ts

```diff
@@ -0,0 +1,176 @@
+import instantsearch from 'instantsearch.js';
+import { connectInfiniteHits } from 'instantsearch.js/es/connectors';
+import { distanceInWords } from 'date-fns';
+import { getCurrentDate } from '../utils';
+
+const currentDate = getCurrentDate();
+
+function createHit(hit, { isHighlighted }) {
+  return `
+<li class="ais-InfiniteHits-item${
+    isHighlighted ? ' infinite-hits-item--highlighted' : ''
+  }">
+  <article class="card">
+    <div class="card-image">
+      <img src="${hit.image}" alt="${hit.title}">
+    </div>
+
+    <div class="card-content">
+      <header>
+        ${
+          hit.topics ? `<span class="card-subject">${hit.topics[0]}</span>` : ''
+        }
+
+        <h1 class="card-title">${instantsearch.highlight({
+          attribute: 'title',
+          hit,
+        })}</h1>
+      </header>
+
+      <p class="card-description">${instantsearch.snippet({
+        attribute: 'description',
+        hit,
+      })}</p>
+
+      <footer>
+        <span class="card-author">${hit.author}</span> – ${distanceInWords(
+    currentDate,
+    hit.date,
+    { addSuffix: true }
+  ).replace('about ', '')}
+      </footer>
+    </div>
+  </article>
+</li>
+`;
+}
+
+function createPlaceholderHit({ isHighlighted }) {
+  return `
+<li class="ais-InfiniteHits-item${
+    isHighlighted ? ' infinite-hits-item--highlighted' : ''
+  }">
+  <article class="card card--placeholder">
+    <div class="card-image">
+    </div>
+
+    <div class="card-content">
+
+    </div>
+  </article>
+</li>
+`;
+}
+
+let globalIsLastPage = false;
+
+const infiniteHits = connectInfiniteHits(
+  ({ results, hits, showMore, isLastPage, widgetParams }, isFirstRender) => {
+    const { container } = widgetParams;
+    const containerNode = document.querySelector(container);
+
+    globalIsLastPage = isLastPage;
```

Comment:
Why do you need `globalIsLastPage`?

Reply:
Since we're in the `isFirstRender` branch, `isLastPage` is only captured once (during `init`) and is therefore always `true` in the intersection condition. We thus need to store it outside of this branch to be able to update it for other renders.

Reply:
I see now, thanks :+1: